### PR TITLE
core.demangle: refactor superfluous mixin use

### PR DESCRIPTION
**Pull request by dkorpel**
See https://github.com/dlang/dmd/pull/16114#pullrequestreview-1851171459

There's a lot more to refactor, but I'm starting small.  I plan to make `errStatus` / `errMsg` fields so a single `error()` function can be used throughout `Demangle`.

### CODE REVIEW

File: druntime/src/core/demangle.d [hunk 1]
```diff
@@ -394,9 +388,15 @@ pure @safe:
...
+        void errorReturn()
+        {
+            errStatus = true;
+        }
+
         if ( 'I' == front )
         {
-            mixin(matchOrF!( "INF" ));
+            if (!match("INF"))
+                return errorReturn();
```
Comment (denizzzka) [inline druntime/src/core/demangle.d:399]: Name is not good: `errorReturn()` actually not returns control flow from `parseReal()` and not returns anything from itself

Comment (denizzzka) [reply druntime/src/core/demangle.d:399]: What about
```d
errStatus = !match("INF");
if (errStatus) return;
```
?

Comment (denizzzka) [reply druntime/src/core/demangle.d:399]: However, your approach also makes sense.

Comment (dkorpel) [reply druntime/src/core/demangle.d]: I don't think it really matters, it's temporary. I just picked something unique so it's easy to rename, but I made it `onError` now.

**Comment by denizzzka** [comment]
> I plan to make errStatus / errMsg fields

I tried to save error messages for succession reason. But in reality error messages are'n used anywhere in the code: all these messages have always been ignored. Perhaps it should be convient to rid of them completely in favor of a boolean flag.

**Comment by dkorpel** [comment]
My idea is to have a single standard `error("msg")` function in `Demangle` that can do whatever it wants with the message: print for debug purposes, discard for release builds, put in a user-specified sink, etc.

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**